Bottom: 69fc9430bd65433269275f6f2143c9e9e548b4f3
Top:    022d0e4473e17ba75c2ba9fbf0d6234ac853c416
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 19:44:55 +0100

draw.c: Split "Gathering" from real drawing routines (Element names)


---

diff --git a/src/draw.c b/src/draw.c
index 75a6c8e..d71982f 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -194,6 +194,23 @@ Redraw (void)
   gui->invalidate_all ();
 }
 
+static void
+draw_element_name (ElementType *element)
+{
+  if ((TEST_FLAG (HIDENAMESFLAG, PCB) && gui->gui) ||
+      TEST_FLAG (HIDENAMEFLAG, element))
+    return;
+  if (doing_pinout || doing_assy)
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else if (TEST_FLAG (SELECTEDFLAG, &ELEMENT_TEXT (PCB, element)))
+    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
+  else if (FRONT (element))
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else
+    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
+  DrawTextLowLevel (&ELEMENT_TEXT (PCB, element), PCB->minSlk);
+}
+
 static int
 name_callback (const BoxType * b, void *cl)
 {
@@ -205,7 +222,7 @@ name_callback (const BoxType * b, void *cl)
     return 0;
 
   if (ON_SIDE (element, *side))
-    DrawElementName (element);
+    draw_element_name (element);
   return 0;
 }
 
@@ -1512,18 +1529,10 @@ DrawElement (ElementTypePtr Element)
 void
 DrawElementName (ElementTypePtr Element)
 {
-  if (gui->gui && TEST_FLAG (HIDENAMESFLAG, PCB))
-    return;
+  assert (Gathering);
+
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
-  if (doing_pinout || doing_assy)
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else if (TEST_FLAG (SELECTEDFLAG, &ELEMENT_TEXT (PCB, Element)))
-    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
-  else if (FRONT (Element))
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else
-    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
   DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
 }
 
@@ -1741,6 +1750,8 @@ EraseElementPinsAndPads (ElementTypePtr Element)
 void
 EraseElementName (ElementTypePtr Element)
 {
+  assert (Gathering);
+
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
   DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
@@ -1839,7 +1850,7 @@ static void
 draw_element (ElementTypePtr element)
 {
   draw_element_package (element);
-  DrawElementName (element);
+  draw_element_name (element);
   DrawElementPinsAndPads (element);
 }
